parsing/lexer.mll

```diff
@@ -190,6 +200,29 @@ let char_for_hexadecimal_code lexbuf i =
   in
   Char.chr (val1 * 16 + val2)
 
+let uchar_for_uchar_escape lexbuf =
+  let len = Lexing.lexeme_end lexbuf - Lexing.lexeme_start lexbuf in
+  let spos = 3 (* skip opening \u{ *) in
+  let epos = len - 2 (* skip closing } *) in
+  let rec hex_num_value acc i =
+    if i > epos then acc else
+    let digit = Char.code (Lexing.lexeme_char lexbuf i) in
+    let value =
```

Comment:
I suggest introduce a 'val hex_digit: char -> int' and use it also in `char_for_hexadecimal_code.  Or perhaps directly 'val hex_from_lexbuf: lexbuf -> int (*pos*)-> int (*len*)-> int`.